mrp_unbuild_tracked_raw_material/models/unbuild.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2019 Akretion (http://www.akretion.com). All Rights Reserved
+# @author David BEAL <david.beal@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as DT_FORMAT
+
+logger = logging.getLogger(__name__)
+
+MESSAGE = "Some of your components are tracked, you have to specify " \
+          "a manufacturing order in order to retrieve " \
+          "the correct components."
+ALTER_MESSAGE = "Alternatively, you may unbuild '%s' tracked product " \
+                "if you set it as '%s'.\n" \
+                "In this case lots'll be automatically created for you."
+
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    def action_unbuild(self):
+        """ We need to catch raise behavior when tracked products
+            are unbuild without an original manufacturing order and
+            go on with another workflow with _action_unbuild_alternative_way()
+        """
+        try:
+            res = super().action_unbuild()
+        except UserError as err:
+            # Unbuild is impossible because of MESSAGE
+            if hasattr(err, 'name') and err.name == _(MESSAGE):
+                if self.product_id.unbuild_purchased:
+                    # In this case it becomes possible to unbuild
+                    return self._action_unbuild_alternative_way()
+                # Here other option to resolve unbuild conditions
+                new_message = _(ALTER_MESSAGE) % (
+                    self.product_id.name, _("Unbuild Purchased"))
+                # We teach user the 2 conditions that make unbuild possible
+                raise UserError(_("%s \n\n%s") % (err.name, new_message))
+            # Here is the Odoo native raise
+            return err
+        return res
+
+    def _action_unbuild_alternative_way(self):
```

Comment:
rename to something like action_unbuild_tracked_product
or bypass_unbuild_tracked_products